discovery_engine_core/ai/ai/src/utils.rs

```diff
@@ -73,6 +75,20 @@ pub(crate) fn system_time_now() -> SystemTime {
     SystemTime::now()
 }
 
+/// Rank documents based on their score.
+///
+/// # Panics
+/// This function will panic if a score for a document is missing.
+pub fn rank<D, S>(documents: &mut [D], scores: &HashMap<D::Id, f32, S>)
+where
+    D: Document,
+    S: std::hash::BuildHasher,
```

Comment:
❓ what's the reason to not use the default for `S`?
```suggestion
pub fn rank<D>(documents: &mut [D], scores: &HashMap<D::Id, f32>)
where
    D: Document,
```

Reply:
It will forbid to use other hashes, and clippy complaints.